docs/publish.py

```diff
@@ -1,41 +1,76 @@
 #!/usr/bin/python3
 import os
 import sys
-__dir__ = os.path.dirname(__file__)
-sys.path.insert(0, os.path.join(__dir__, ".."))
+import typing as ty
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+sys.path.insert(0, os.path.join(script_dir, ".."))
 
 import sphinx.cmd.build
 import ipfshttpclient
 
-# Ensure working directory is script directory
-os.chdir(__dir__)
+os.chdir(script_dir)
+
+
+def main(argv: ty.List[str]) -> int:
+	if len(argv) >= 1:
+		ipns_key = argv[0]
+	else:
+		ipns_key = None
 
-def main(argv=sys.argv[1:], program=sys.argv[0]):
-	if len(argv) != 1:
-		print("Usage: {0} [IPNS-key]".format(os.path.basename(program)))
+		print("Usage: {0} [IPNS-key]".format(os.path.basename(__file__)))
 		print()
 		print("!! Continuing without publishing to IPNS !!")
 		print()
-	
+
+	ipfs_api_address: str = os.getenv('IPFS_API_MULTI_ADDR', str(ipfshttpclient.DEFAULT_ADDR))
+	ipfs_api_username: ty.Optional[str] = os.getenv('IPFS_API_USERNAME', None)
+	ipfs_api_password: ty.Optional[str] = os.getenv('IPFS_API_PASSWORD', None)
```

Comment:
Using environment variables seems weird to me here. This would make a lot more sense as command-line parameters instead?

Reply:
Using environment variables for credentials is very common (see Docker).

Reply:
It feels wrong to do it this way, but I get your point. Could you drop `IPFS_API_MULTI_ADDR` in favour of `PY_IPFS_HTTP_CLIENT_DEFAULT_ADDR` and add a new `PY_IPFS_HTTP_CLIENT_DEFAULT_AUTH` instead of the `IPFS_API_USERNAME` and `IPFS_API_PASSWORD` fields?

At least that way passing the envs this way not be special to just this specific piece of maintenance code.